Arrowgene.Ddon.GameServer/Handler/CraftStartEquipGradeUpHandler.cs

```diff
@@ -0,0 +1,273 @@
+            uint EquipRank = json_data.Unk0;
+            bool canContinue = true;
+            bool IsGreatSuccess = _random.Next(5) == 0; // 1 in 5 chance to be true, someone said it was 20%.
+            bool DoUpgrade = false;
+
+            uint currentTotalEquipPoint = equipItem.EquipPoints;
+            uint previousTotalEquipPoint = currentTotalEquipPoint;
+            uint addEquipPoint = 0;     
+            double minMultiplier = 0.8;
+            double maxMultiplier = 1.2;
+            double pointsMultiplier = minMultiplier + (_random.NextDouble() * (maxMultiplier - minMultiplier));
+
+            // TODO: Figure out if you can upgrade several times in a single interaction and if so, handle that lol
+            List<CDataCommonU32> gradeuplist = new List<CDataCommonU32>()
+            {
+                new CDataCommonU32(gearupgradeID)
+            };
+            CDataEquipSlot EquipmentSlot = new CDataEquipSlot()
+            {
+                CharacterId = 0,
+                PawnId = 0,
+                EquipType = 0,
+                EquipSlotNo = 0
+            };
+            CDataCurrentEquipInfo CurrentEquipInfo = new CDataCurrentEquipInfo()
+            {
+                ItemUId = equipItemUID,
+                EquipSlot = EquipmentSlot
+            };
+            // More dummy data, looks like its DragonAugment related.
+            CDataCraftStartEquipGradeUpUnk0Unk0 DragonAugmentData = new CDataCraftStartEquipGradeUpUnk0Unk0()
+            {
+                Unk0 = 0,          // Probably DragonAugment related.
+                Unk1 = 0,
+                Unk2 = 0,          // setting this to a value above 0 seems to stop displaying "UP" ?
+                Unk3 = 0,          // displays "UP" next to the DA upon succesful enhance.
+                IsMax = false,      // displays Max on the DA popup.
+            };
+            CDataCraftStartEquipGradeUpUnk0 dummydata = new CDataCraftStartEquipGradeUpUnk0()
+            {
+                Unk0 = new List<CDataCraftStartEquipGradeUpUnk0Unk0> { DragonAugmentData },
+                Unk1 = 0,
+                Unk2 = 0,
+                Unk3 = 0,               // No idea what these 3 bytes are for
+                DragonAugment = false,    // makes the DragonAugment slot popup appear if set to true.
+            };
+            // TODO: Source these values accurately when we know what they are. ^
+
+            List<CDataItemUpdateResult> updateResults;
+            var res = new S2CCraftStartEquipGradeUpRes();
+            S2CItemUpdateCharacterItemNtc updateCharacterItemNtc = new S2CItemUpdateCharacterItemNtc();
+
+            #endregion
+
+            #region Point Distribution 
+            // Handles adding EquipPoints.
+            // These values for adding equippoints are completely made up and not representative of any real Pawn craft levels.
+            if(IsGreatSuccess == true)
+            {
+                addEquipPoint = 300;
```

Comment:
Should these crafting point values be sourced from some json which can be updated later?

Reply:
At the very least, move them to a named constant with a TODO comment